erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -859,3 +861,10 @@ frappe.ui.form.on('Payment Entry Deduction', {
 		frm.events.set_unallocated_amount(frm);
 	}
 })
+frappe.ui.form.on('Payment Entry', {
```

Comment:
As per my above comment, if balance comes as per total outstanding, this piece of code is not needed, I think. Also  this will cause asynchronous issue, as there are multiple ajax call involved.

Reply:
>As per my above comment, if balance comes as per total outstanding, this piece of code is not needed, I think. 

This is required to show party balance of selected cost center. If no cost center is selected, then show overall balance at company level.

>Also this will cause asynchronous issue, as there are multiple ajax call involved.

Agreed. Will enhance this.

Reply:
Again, I think if cost centre is selected, perhaps show cost centre and total for entire company.